tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/MultipartUtilities.cs

```diff
@@ -431,6 +433,133 @@ private static void RemoveCurrentBoundary(
             }
         }
 
+        /// <summary>
+        /// Normalizes file paths in Content-Disposition header values by converting Windows-style backslashes to forward slashes.
+        /// This ensures cross-platform compatibility when comparing multipart/form-data requests.
+        /// </summary>
+        /// <param name="value">The Content-Disposition header value to normalize.</param>
+        /// <returns>The normalized Content-Disposition header value with forward slashes in filenames.</returns>
+        public static string NormalizeFilenameFromContentDispositionValue(string value)
+        {
+            if (string.IsNullOrEmpty(value) || !value.Contains("filename"))
+            {
+                return value;
+            }
+
+            var spanValue = value.AsSpan();
+            var semicolonIndex = spanValue.IndexOf(';');
+            if (semicolonIndex == -1) // no header parameters
+            {
+                return value;
+            }
+
+            var outputValue = new StringBuilder(value.Length + 5);
+            var dispositionType = spanValue[..(semicolonIndex + 1)];
+            outputValue.Append(dispositionType); // disposition type with ;
+
+            var remaining = spanValue[(semicolonIndex + 1)..];
+
+            // parse each parameter
+            while (!remaining.IsEmpty)
+            {
+                // leading white space
+                int nonWhitespace = 0;
+                while (nonWhitespace < remaining.Length && char.IsWhiteSpace(remaining[nonWhitespace]))
+                {
+                    nonWhitespace++;
+                }
+                outputValue.Append(remaining[..nonWhitespace]);
+                remaining = remaining[nonWhitespace..];
+
+                // get param slice: "name=value" and param name and value
+                var nextSemicolon = remaining.IndexOf(';');
+                var param = nextSemicolon == -1 ? remaining : remaining.Slice(0, nextSemicolon + 1);
+                var equalIndex = param.IndexOf('=');
+                var paramName = equalIndex == -1 ? param : param.Slice(0, equalIndex);
+                var paramValue = equalIndex == -1 ? ReadOnlySpan<char>.Empty : param.Slice(equalIndex + 1);
+
+                if (paramName.SequenceEqual("filename".AsSpan()))
```

Comment:
I do not like having specific code for handling `filename` vs `filename*`.

I can understand normalizing paths generally within a `Content-Disposition` header, but having the specific code handling for the specific param seems like a bad pattern. It's very specific to this specific param, instead of being a bit more general.

Is there anything stopping us from normalizing backslashes in the Content-Disposition header generally?

Reply:
I can see exactly why you implemented this way, and I'm not convinced that this is the right solution. It feels extremely tightly bound to an azure-specific problem.